app/uk/gov/hmrc/tradergoodsprofilesrouter/models/request/WithdrawReasonRequest.scala

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tradergoodsprofilesrouter.models.request
+
+import play.api.libs.json.{Json, OFormat}
+
+case class WithdrawReasonRequest(withdrawReason: Option[String])
+
+object WithdrawReasonRequest {
+  implicit val format: OFormat[WithdrawReasonRequest] = Json.format[WithdrawReasonRequest]
```

Comment:
does this work if caller wants to remove the withdrawReason? like if caller passes `{withdrawReason: null}` to remove the same? we currently have similar issue for update record and maintain profile where calls wants to reset some of the optional fields to null but it is not working as null fields in request are getting omitting.

Reply:
This should never happens. The withdraw reason should never be null from the trader-goods-profiles. This should be either an empty json (if not present) or a valid json if present. If I don't mistaken we have the problem with null in json only in the EIS response because they wrongly send null as value json